third_party/move/move-model/src/builder/exp_builder.rs

```diff
@@ -2868,11 +2848,29 @@ impl<'env, 'translator, 'module_translator> ExpTranslator<'env, 'translator, 'mo
     ) -> Result<(), TypeUnificationError> {
         for (idx, ctr) in constraints {
             let ty = &args[*idx];
-            subs.eval_constraint(loc, ty, WideningOrder::LeftToRight, ctr.to_owned())?;
+            subs.eval_constraint(
+                &self.unification_context,
+                loc,
+                &subs.specialize(ty),
+                WideningOrder::LeftToRight,
+                ctr.to_owned(),
+            )?;
         }
         Ok(())
     }
 
+    fn add_constraint_and_report(&mut self, loc: &Loc, ty: &Type, c: Constraint) {
```

Comment:
The function above has a comment that effectively says "adds constraints and reports errors", but it is this function that does that.

Reply:
Done